### PR TITLE
Readme

### PR DESCRIPTION
**Pull request by mbrandonw**
basic readme

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,69 @@
...
+
+A type to express a value that holds one of two other types.
+
+```swift
+import Either
+
+let intOrString = Either<Int, String>.left(2)
+
+intOrString
+  |> bimap({ $0 + 1 })({ $0 + "!" }) // => .left(3)
```
Comment (stephencelis) [inline README.md]: Our `bimap` (and `either` and `validate`) _methods_ have been allowing for two args to make this curried sitch a little less awkward, so I think we could write:

```
intOrString
  .bimap({ $0 + 1 }, { $0 + "!" })
```

File: README.md [hunk 2]
```diff
@@ -1,5 +1,69 @@
...
+```
+
+## `Optics`
+
+A `Lens` type and a bridge between the lens world and the Swift keypath world.
```
Comment (stephencelis) [inline README.md]: Should probably be "key path" or `KeyPath`: https://github.com/apple/swift-evolution/blob/master/proposals/0161-key-paths.md

File: README.md [hunk 3]
```diff
@@ -1,5 +1,69 @@
...
+
+```swift
+import Optics
```
Comment (stephencelis) [inline README.md]: Think we need to `import Prelude` for this snippet to work?

File: Tests/OpticsTests/OpticsTests.swift [hunk 4]
```diff
@@ -21,7 +21,7 @@ class PreludeTests: XCTestCase {
   func testLens() {
     let user = User(id: 1, name: "Stephen")
     let overUser = user
-      |> (\.id) .~ 2
+      |> \.id .~ 2
```
Comment (stephencelis) [inline Tests/OpticsTests/OpticsTests.swift]: Parens used to be required because of ambiguity with `.~`. Does this work now???

Comment (mbrandonw) [inline Tests/OpticsTests/OpticsTests.swift]: it does!

**Pull request merged**